### ISSUE TITLE
Stop training the model based on an error bound

### ISSUE THREAD

**Issue by shanmukhi1998** [lululxvi/deepxde#1116]
Hello Lu,

Thanks for developing such a wonderful module on PINNs. I am solving an **inverse** problem in PINNs. I am recording the losses every 1000 iterations/epochs. I would like to stop training the model once the loss is less than some value, say 10^-4. How may I do that? So far, I have been training the model for a fixed number of iterations and I am waiting till those number of iterations are done. 
For that matter, what are the different ways to automatically stop training the model based on certain conditions?
It would be of great help to me if you could clarify this.

Thanks,
Shanmukhi

**Comment by alabaykazakh** [comment]
```
    early_stopping = dde.callbacks.EarlyStopping(min_delta=1e-4, patience=2000)
    model.compile("adam", lr=1e-3)
    model.train(epochs=10000, disregard_previous_best=True, callbacks=[early_stopping])
```

**Comment by shanmukhi1998** [comment]
Hi @alabaykazakh . Thank you for answering the question. Just one last clarification. In the context of inverse problem of PINNs, this min_delta = 1e-4 means, even if one of the losses (training/physics) reaches 10^-4, the training will stop, right? Or does it refer to just the training data loss?

**Comment by alabaykazakh** [comment]
> EarlyStopping

![image](https://user-images.githubusercontent.com/61251090/213170018-5a53345a-02bd-4729-a74c-a7e4af56a833.png)
You can define only one of the losses by adding the argument `monitor` to the callback. For more information, please read the code documentation https://deepxde.readthedocs.io/en/latest/_modules/deepxde/callbacks.html#EarlyStopping.

**Comment by alabaykazakh** [comment]
> ```
>     early_stopping = dde.callbacks.EarlyStopping(min_delta =1000, patience=2000, baseline=1e-4 )
>     model.compile("adam", lr=1e-3)
>     model.train(epochs=10000, disregard_previous_best=True, callbacks=[early_stopping])
> ```

As far as I understood: I defined the `min_delta` high, so when you reach the `baseline` of 1e-4, probably your improvement will be smaller than this `min_delta` , therefore your training stops.

**Comment by shanmukhi1998** [comment]
Thanks for your inputs @alabaykazakh . But if losses are in the range of 1e-3, min_delta does not really seem to make sense. Probably, I would want my loss to go to at most 1e-4.
